src/loci/formats/in/ZarrReader.java

```diff
@@ -524,18 +660,83 @@ else if (multiscaleAxes.get(i) instanceof HashMap) {
+            int resolutionCount = (Integer)(uniqueResCounts.toArray())[0];
+            for (int j = 0; j < resolutionCount; j++) {
+              String key = rowName + File.separator + columnName + File.separator + i + File.separator + j;
+              if (Files.isDirectory(Paths.get(canonicalPath+File.separator+key))) {
+                arrayPaths.add(rowName + File.separator + columnName + File.separator + i + File.separator + j);
+              }
+              else {
+                LOGGER.debug("Skipping array path as sparse data: {}", key);
+              }
+            }
+          }
+        }
+      }
+    }
+  }
+
+  private void generateGroupKeys(Map<String, Object> attr, String canonicalPath) {
+    Map<Object, Object> plates = (Map<Object, Object>) attr.get("plate");
+    if (plates != null) {
+      ArrayList<Object> columns = (ArrayList<Object>)plates.get("columns");
+      ArrayList<Object> rows = (ArrayList<Object>)plates.get("rows");
+      Integer fieldCount = (Integer) plates.get("field_count");
+
+      for (Object row: rows) {
+        String rowName = ((Map<String, String>) row).get("name");
+        if (Files.isDirectory(Paths.get(canonicalPath+File.separator+rowName))) {
+          groupKeys.add(rowName);
+        }
+        else {
+          LOGGER.debug("Skipping group key as sparse data: {}", rowName);
+        }
+        for (Object column: columns) {
+          String columnName = ((Map<String, String>) column).get("name");
+          String columnKey = rowName + File.separator + columnName;
+          if (Files.isDirectory(Paths.get(canonicalPath+File.separator+columnKey))) {
+            groupKeys.add(columnKey);
+          }
+          else {
+            LOGGER.debug("Skipping group key as sparse data: {}", columnKey);
+          }
+          for (int i = 0; i < fieldCount; i++) {
+            String key = rowName + File.separator + columnName + File.separator + i;
+            if (Files.isDirectory(Paths.get(canonicalPath+File.separator+key))) {
+              groupKeys.add(key);
+            }
+            else {
+              LOGGER.debug("Skipping group key as sparse data: {}", key);
+            }
+          }
+        }
+      }
+    }
+  }
+
+  private void parsePlate(Map<String, Object> attr, String root, String key, MetadataStore store) throws IOException, FormatException {
     Map<Object, Object> plates = (Map<Object, Object>) attr.get("plate");
     if (plates != null) {
       ArrayList<Object> columns = (ArrayList<Object>)plates.get("columns");
       ArrayList<Object> rows = (ArrayList<Object>)plates.get("rows");
       ArrayList<Object> wells = (ArrayList<Object>)plates.get("wells");
       ArrayList<Object>  acquisitions = (ArrayList<Object> )plates.get("acquisitions");
       String plateName = (String) plates.get("name");
-      String fieldCount = (String) plates.get("filed_count");
+      Integer fieldCount = (Integer) plates.get("field_count");
```

Comment:
This isn't directly related to this PR, right?

Reply:
Not directly related, it fixes a bug that was discovered during this round of testing